test/unit/math/prim/scal/fun/distance_test.cpp

```diff
@@ -0,0 +1,39 @@
+#include <stan/math/prim/scal.hpp>
+#include <gtest/gtest.h>
+
+TEST(MathFunctions, distance) {
```

Comment:
@bob-carpenter You mentioned that you'd done some automated tests for higher order autodiff simple functions? Is there a particular example for doing that? I remember seeing something on the discourse but didn't find it with a couple searches. Any chance it would apply here?

Reply:
Thanks for these tips. It will take more time for me to get acquainted with these metaprogramming aspects, but it is definitely useful thing to do. Then I'll see what I can do to reduce the length of tests.

Reply:
I'm sure there are ways to reduce the size of the tests using metaprogramming, but I'd rather tackle that problem in a separate pull request.

Reply:
Not really part of the review, but I'd be interested in compiling these into a test suite for the GP covariance functions, as I already have a working template. Can someone can point me automated tests? or may be @rtrangucci can point me to references regarding:

> ways to reduce the size of the tests using metaprogramming

 There are also tests in here that `cov_exp_quad` (`gp_exp_quad_cov`) does not contain, such as checking gradients and whether `var`'s are on the stack, that I need to include in `cov_exp_quad` and other covariance functions. I'll open an issue about it soon.